src/test/java/api/loans/scenarios/CheckInDeclaredLostItemTest.java

```diff
@@ -316,14 +332,105 @@ public void shouldNotChargeOverdueFineWhenNotStatedByPolicy() {
+
+  @Test
+  public void shouldRefundOnlyLastLoanForLostAndPaidItem() {
+    final double firstFee = 20.00;
+    final double secondFee = 30.00;
+
+    useChargeableRefundableLostItemFee(firstFee, 0.0);
+
+    final UUID firstLoanId = declareItemLost();
+    mockClockManagerToReturnFixedDateTime(now(DateTimeZone.UTC).plusMinutes(2));
+    // Item fee won't be cancelled, because refund period is exceeded
+    checkInFixture.checkInByBarcode(item);
+    assertThat(itemsClient.getById(item.getId()).getJson(), isAvailable());
+
+    final UUID secondLoanId = declareItemLostAndPayFees(secondFee);
+    checkInFixture.checkInByBarcode(item);
+
+    verifyLostItemFeeAccount(firstLoanId, allOf(
+      hasJsonPath("status.name", "Open"),
+      hasJsonPath("remaining", firstFee),
+      hasJsonPath("amount", firstFee)));
+    verifyLostItemFeeAccountAction(firstLoanId, not(isCloseActionCreated(firstFee)));
+
+    verifyLostItemFeeAccount(secondLoanId, isAccountRefundedFully(secondFee));
+    verifyLostItemFeeAccountAction(secondLoanId, arePaymentRefundActionsCreated(secondFee));
+  }
+
+  @Test
+  public void shouldFailIfNoLoanForLostAndPaidItem() {
+    final double setCost = 20.00;
+
+    declareItemLostAndPayFees(setCost);
+
+    // Remove the loan from storage
+    loansFixture.deleteLoan(loan.getId());
+
+    final Response checkInResponse = checkInFixture.attemptCheckInByBarcode(
+      new CheckInByBarcodeRequestBuilder()
+        .at(servicePointsFixture.cd1())
+        .forItem(item));
+
+    assertThat(checkInResponse.getJson(), hasErrorWith(allOf(
+      hasMessage("Item is lost however there is no declared lost loan found"),
+      hasParameter("itemId", item.getId().toString()))));
+  }
+
+  @Test
+  public void shouldFailIfLastLoanIsNotDeclaredLostForLostAndPaidItem() {
+    loan = checkOutFixture.checkOutByBarcode(item, usersFixture.jessica());
+
+    checkInFixture.checkInByBarcode(item);
+
+    // Update the item status in storage
+    itemsClient.replace(item.getId(), itemsClient.get(item).getJson().copy()
+      .put("status", new JsonObject().put("name", "Lost and paid")));
+
+    final Response checkInResponse = checkInFixture.attemptCheckInByBarcode(
+      new CheckInByBarcodeRequestBuilder()
+        .at(servicePointsFixture.cd1())
+        .forItem(item));
+
+    assertThat(checkInResponse.getJson(), hasErrorWith(allOf(
+      hasMessage("Last loan for lost item is not declared lost"),
+      hasParameter("loanId", loan.getId().toString()))));
+  }
+
+  private UUID declareItemLostAndPayFees(double setCostFeeAmount) {
+    useChargeableRefundableLostItemFee(setCostFeeAmount, 0);
+
+    final UUID loanId = declareItemLost();
+
+    feeFineAccountFixture.payLostItemFee(loanId);
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loanId);
+
+    assertThat(itemsClient.getById(item.getId()).getJson(), isLostAndPaid());
+    assertThat(loansFixture.getLoanById(loanId).getJson(), isClosed());
+
+    return loanId;
+  }
```

Comment:
Could this be split into two methods? I think it is easier to understand if the reader can tell that declaring an item lost and paying the associated fees are two separate steps in the process. What do you think?

Reply:
Split into 
```
declareItemLost(setCostFee);
resolveLostItemFee();
```